### PR TITLE
Feature/te 612 ota api

### PR DESCRIPTION
**Pull request by patrickAnyline**

### CODE REVIEW

File: plugin/android/src/main/java/com/anyline/reactnative/updateAsset/AssetContextJsonParser.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @Nullable
+    public AssetContext parseJson(Context context, String json) throws JSONException {
+        JSONObject jsonObject = new JSONObject(json);
+
+        if (jsonObject.has("projectID")) {
+            return new ProjectContext(context, jsonObject);
+        } else if (jsonObject.has("product")) {
+            return new ProductContext(context, jsonObject);
+        }
+        return null;
```
Comment (peterAnyline) [inline plugin/android/src/main/java/com/anyline/reactnative/updateAsset/AssetContextJsonParser.java:25]: maybe throw an exception when neither is provided?

File: example/RNExampleApp/config/OtaConfig.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
...VSZXBvcnRDYWNoaW5nIjogZmFsc2UsCiAgIm1ham9yVmVyc2lvbiI6ICIyNSIsCiAgIm1heERheXNOb3RSZXBvcnRlZCI6IDUsCiAgImFkdmFuY2VkQmFyY29kZSI6IHRydWUsCiAgIm11bHRpQmFyY29kZSI6IHRydWUsCiAgInN1cHBvcnRlZEJhcmNvZGVGb3JtYXRzIjogWwogICAgIkFMTCIKICBdLAogICJwaW5nUmVwb3J0aW5nIjogdHJ1ZSwKICAicGxhdGZvcm0iOiBbCiAgICAiaU9TIiwKICAgICJBbmRyb2lkIgogIF0sCiAgInNjb3BlIjogWwogICAgIkFMTCIKICBdLAogICJzaG93UG9wVXBBZnRlckV4cGlyeSI6IGZhbHNlLAogICJzaG93V2F0ZXJtYXJrIjogdHJ1ZSwKICAidG9sZXJhbmNlRGF5cyI6IDkwLAogICJ2YWxpZCI6ICIyMDIxLTA2LTMwIiwKICAiaW9zSWRlbnRpZmllciI6IFsKICAgICJjb20uYW55bGluZS5leGFtcGxlLnJlYWN0bmF0aXZlIiwKICAgICJjb20uYW55bGluZS5leGFtcGxlLnJlYWN0bmF0aXZlLmJldGEiCiAgXSwKICAiYW5kcm9pZElkZW50aWZpZXIiOiBbCiAgICAiY29tLmFueWxpbmUuZXhhbXBsZS5yZWFjdG5hdGl2ZSIKICBdCn0KWFdnaUNoTmZxL3hEOWtqSmVqQW14dFNSS2x6T2I2a3A2SEFxcGNvS2pjQ0pUSE1sNFVpRUJCdzFyckJyZHdTU2lycmovaWROUTZEMnFwUWZNcFZ6OU4reFlqSVJ3bmRzdWxtNVhEMFhzZ2srTVI3a3BiSXhmOVhPRHpHNzVmWkZjdGhkNUh3eGtxWnJEVy9aZW1OOHlsdGtaRkpSNzhOYzV3REw3TFZmZCtzUi9XNDM3b05EVkVjaS9MRVBXVEhJd0p0VG80ZmJnSHhoMGV6QzFTVzVPVFZnL3FheUlDeVVnNVNIVEtXR0ZzcmJsR3ZBNjc1OHc2YnBmM0xzeFVmTitXYWZIY1dmNmtORHdpTDVXZ2U2QWN0Q3k2RW9KeTFuRFZEUVVBMFhQV2NYMzZHakVHeFJxZGN5YW1QcUpWNDNDWjN2QUtHUFZxSnhrUHFKNTFId2JBPT0=',
```
Comment (peterAnyline) [inline example/RNExampleApp/config/OtaConfig.js:2]: please use placeholder values only ;)

**Review by peterAnyline — COMMENTED** [review]
don't expose real project credentials, throw exception when neither projectID nor product is provided

File: plugin/android/src/main/java/com/anyline/reactnative/updateAsset/AssetContextJsonParser.java (hunk 1 above)
Comment (patrickAnyline) [reply plugin/android/src/main/java/com/anyline/reactnative/updateAsset/AssetContextJsonParser.java]: done

**Pull request merged**